leftwm-core/src/models/window_change.rs

```diff
@@ -59,7 +59,8 @@ impl WindowChange {
         if let Some(urgent) = self.urgent {
             let changed_urgent = window.urgent != urgent;
             changed = changed || changed_urgent;
-            window.urgent = urgent;
+            // If the window is already visible, there is no need to mark it as urgent
+            window.urgent = urgent && !window.visible();
```

Comment:
Good catch! That should be at least one bug removed from the urgent-hint implementation. I'll look into it if the workaround mentioned above is still needed after this one.